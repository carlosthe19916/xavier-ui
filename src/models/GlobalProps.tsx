export interface GlobalProps {
    history: any;
    location: any;
    match: any;
}
