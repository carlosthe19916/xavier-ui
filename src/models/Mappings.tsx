export interface FlagAssessmentModel {
    flag: string;
    label: string;
    flagAssessmentOSModels: FlagAssessmentOSModel[]
}

export interface FlagAssessmentOSModel {
    osName: string | undefined;
    assessment: string;
}
