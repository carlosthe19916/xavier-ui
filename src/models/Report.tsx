export interface Report {
    id: number;
    customerId: string;
    fileName: string;
    numberOfHosts: number;
    totalDiskSpace: number;
    totalPrice: number;
    creationDate: number
}
