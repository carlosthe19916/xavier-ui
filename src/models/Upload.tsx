export interface Upload {
    file: File,
    success: boolean | null,
    error: string | null,
    progress: number,
    uploading: boolean
}
