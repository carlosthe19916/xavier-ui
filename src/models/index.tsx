export * from './Report';
export * from './Upload';
export * from './User';
export * from './Mappings';
