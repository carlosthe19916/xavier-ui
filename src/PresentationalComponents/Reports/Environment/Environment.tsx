import React, { Component } from 'react';
import {
    Table,
    TableBody,
    TableVariant
} from '@patternfly/react-table';
import { EnvironmentModel } from '../../../models';
import { isNullOrUndefined } from '../../../Utilities/formUtils';
import { formatNumber } from '../../../Utilities/formatValue';

interface Props {
    data: EnvironmentModel
}

interface State {
}

class Environment extends Component<Props, State> {

    constructor(props: Props) {
        super(props);
    }

    public render() {
        const { data: {
            hypervisors,
            year1Hypervisor,
            year2Hypervisor,
            year3Hypervisor,
            growthRatePercentage
        }} = this.props;

        const columns = [
            '',
            '',
            ''
        ];

        const rows = [
            [
                'ESXi hypervisors (2-socket servers)',
                '',
                isNullOrUndefined(hypervisors) ? 'Unknown' : formatNumber(hypervisors, 0)
            ],
            [
                'ESXi hypervisors would be migrated to RH technologies',
                'Year1 (incremental)',
                isNullOrUndefined(year1Hypervisor) ? 'Unknown' : formatNumber(year1Hypervisor, 0)
            ],
            [
                '',
                'Year2 (incremental)',
                isNullOrUndefined(year2Hypervisor) ? 'Unknown' : formatNumber(year2Hypervisor, 0)
            ],
            [
                '',
                'Year3 (incremental)',
                isNullOrUndefined(year3Hypervisor) ? 'Unknown' : formatNumber(year3Hypervisor, 0)
            ],
            [
                'Year-over-year growth rate for new hypervisors',
                '',
                isNullOrUndefined(growthRatePercentage) ? 'Unknown' : `${(growthRatePercentage * 100).toLocaleString()}%`
            ]
        ];

        return (
            <React.Fragment>
                <Table aria-label='Environment'
                    cells={ columns }
                    rows={ rows }
                    variant={ TableVariant.compact }
                >
                    <TableBody />
                </Table>
            </React.Fragment>
        );
    }
}

export default Environment;
