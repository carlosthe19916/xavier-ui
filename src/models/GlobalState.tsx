import { Report } from './Report';
import { Upload } from './Upload';

export interface GlobalState {
    reports: {
        report: Report,
        reports: Report[];
        loading: boolean;
        error: string;
        total: number;
    },
    uploads: {
        uploads: Upload[]
    }
}
