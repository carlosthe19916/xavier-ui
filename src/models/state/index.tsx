import { Report } from '../Report';
import { Upload } from '../Upload';

export interface GlobalState {
    reports: ReportState,
    uploads: UploadState
}

export interface ReportState {
    error: string | null;
    total: number;
    report: Report | null;
    reports: Report[];
    loading: boolean;
}

export interface UploadState {
    uploads: Upload[]
}