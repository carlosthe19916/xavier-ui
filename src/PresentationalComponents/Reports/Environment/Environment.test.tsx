import { shallow } from "enzyme";
import Environment from "./Environment";
import { EnvironmentModel } from "../../../models";

const baseData: EnvironmentModel = {
  sourceProductIndicator: 0,
  hypervisors: 1,
  year1Hypervisor: 12,
  year2Hypervisor: 123,
  year3Hypervisor: 1234,
  growthRatePercentage: 0.05
};

describe("Environment", () => {
  it("expect to render", () => {
    const data = {
      ...baseData
    };

    const wrapper = shallow(
      <Environment data={ data } />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("expect to render using null and undefined data", () => {
    const data = {
      ...baseData,
      hypervisors: null,
      year1Hypervisor: undefined,
      year2Hypervisor: null,
      year3Hypervisor: undefined,
      growthRatePercentage: null
    };

    const wrapper = shallow(
      <Environment data={ data } />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("expect to render using values equal to 0", () => {
    const data = {
      ...baseData,
      hypervisors: 0,
      year1Hypervisor: 0,
      year2Hypervisor: 0,
      year3Hypervisor: 0,
      growthRatePercentage: 0
    };

    const wrapper = shallow(
      <Environment data={ data } />
    );

    expect(wrapper).toMatchSnapshot();
  });

});
