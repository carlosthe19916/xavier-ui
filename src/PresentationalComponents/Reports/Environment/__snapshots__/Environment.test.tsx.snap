// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Environment expect to render 1`] = `
<Fragment>
  <Table
    aria-label="Environment"
    borders={true}
    cells={
      Array [
        "",
        "",
        "",
      ]
    }
    className=""
    contentId="expanded-content"
    dropdownDirection="down"
    dropdownPosition="right"
    expandId="expandable-toggle"
    gridBreakPoint="grid-md"
    onCollapse={null}
    onExpand={null}
    role="grid"
    rowLabeledBy="simple-node"
    rows={
      Array [
        Array [
          "Licenses for ESXi Hypervisors (2 socket)",
          "",
          "1",
        ],
        Array [
          "VMware licenses (2 socket) to migrate to RH technologies",
          "Year1 (incremental)",
          "12",
        ],
        Array [
          "",
          "Year2 (incremental)",
          "123",
        ],
        Array [
          "",
          "Year3 (incremental)",
          "1,234",
        ],
        Array [
          "Year-over-year growth rate for new hypervisors",
          "",
          "5%",
        ],
      ]
    }
    variant="compact"
  >
    <TableBody
      className=""
      onRowClick={[Function]}
      rowKey="id"
    />
  </Table>
</Fragment>
`;

exports[`Environment expect to render using null and undefined data 1`] = `
<Fragment>
  <Table
    aria-label="Environment"
    borders={true}
    cells={
      Array [
        "",
        "",
        "",
      ]
    }
    className=""
    contentId="expanded-content"
    dropdownDirection="down"
    dropdownPosition="right"
    expandId="expandable-toggle"
    gridBreakPoint="grid-md"
    onCollapse={null}
    onExpand={null}
    role="grid"
    rowLabeledBy="simple-node"
    rows={
      Array [
        Array [
          "Licenses for ESXi Hypervisors (2 socket)",
          "",
          "Unknown",
        ],
        Array [
          "VMware licenses (2 socket) to migrate to RH technologies",
          "Year1 (incremental)",
          "Unknown",
        ],
        Array [
          "",
          "Year2 (incremental)",
          "Unknown",
        ],
        Array [
          "",
          "Year3 (incremental)",
          "Unknown",
        ],
        Array [
          "Year-over-year growth rate for new hypervisors",
          "",
          "Unknown",
        ],
      ]
    }
    variant="compact"
  >
    <TableBody
      className=""
      onRowClick={[Function]}
      rowKey="id"
    />
  </Table>
</Fragment>
`;

exports[`Environment expect to render using null and undefined data 2`] = `
<Fragment>
  <Table
    aria-label="Environment"
    borders={true}
    cells={
      Array [
        "",
        "",
        "",
      ]
    }
    className=""
    contentId="expanded-content"
    dropdownDirection="down"
    dropdownPosition="right"
    expandId="expandable-toggle"
    gridBreakPoint="grid-md"
    onCollapse={null}
    onExpand={null}
    role="grid"
    rowLabeledBy="simple-node"
    rows={
      Array [
        Array [
          "Licenses for ESXi Hypervisors (2 socket)",
          "",
          "Unknown",
        ],
        Array [
          "VMware licenses (2 socket) to migrate to RH technologies",
          "Year1 (incremental)",
          "Unknown",
        ],
        Array [
          "",
          "Year2 (incremental)",
          "Unknown",
        ],
        Array [
          "",
          "Year3 (incremental)",
          "Unknown",
        ],
        Array [
          "Year-over-year growth rate for new hypervisors",
          "",
          "Unknown",
        ],
      ]
    }
    variant="compact"
  >
    <TableBody
      className=""
      onRowClick={[Function]}
      rowKey="id"
    />
  </Table>
</Fragment>
`;

exports[`Environment expect to render using values equal to 0 1`] = `
<Fragment>
  <Table
    aria-label="Environment"
    borders={true}
    cells={
      Array [
        "",
        "",
        "",
      ]
    }
    className=""
    contentId="expanded-content"
    dropdownDirection="down"
    dropdownPosition="right"
    expandId="expandable-toggle"
    gridBreakPoint="grid-md"
    onCollapse={null}
    onExpand={null}
    role="grid"
    rowLabeledBy="simple-node"
    rows={
      Array [
        Array [
          "Licenses for ESXi Hypervisors (2 socket)",
          "",
          "0",
        ],
        Array [
          "VMware licenses (2 socket) to migrate to RH technologies",
          "Year1 (incremental)",
          "0",
        ],
        Array [
          "",
          "Year2 (incremental)",
          "0",
        ],
        Array [
          "",
          "Year3 (incremental)",
          "0",
        ],
        Array [
          "Year-over-year growth rate for new hypervisors",
          "",
          "0%",
        ],
      ]
    }
    variant="compact"
  >
    <TableBody
      className=""
      onRowClick={[Function]}
      rowKey="id"
    />
  </Table>
</Fragment>
`;
